spec/lib/miq_ldap_spec.rb

```diff
@@ -119,4 +119,20 @@
     data = "\001\005\000\000\000\000\000\005\025\000\000\000+\206\301\364y\307\r\302=\336p\216\237\004\000\000"
     expect(MiqLdap.sid_to_s(data)).to eq("S-1-5-21-4106323499-3255682937-2389761597-1183")
   end
+
+  it 'returns a hostname when a hostname is availble and does not disable verify mode' do
+    allow(TCPSocket).to receive(:gethostbyname).and_return(["testhostname", "aliases", "type", "192.168.252.20"])
+    allow(TCPSocket).to receive(:new)
+    ldap = MiqLdap.new(:host => ["testhostname", "localhost", "dummy", @host])
+    expect(ldap.ldap.host).to eq("testhostname")
+    expect(ldap.ldap.inspect).to_not include("@encryption={:tls_options=>{:verify_mode=>0}}")
```

Comment:
Would it be more reliable to do the following?
```ruby
expect(ldap.ldap.instance_variable_get(:@encryption)).to be_nil
```

Reply:
[Line 51](https://github.com/ManageIQ/manageiq/pull/16850/files#diff-b2b4e22924785a504e3b054fec263056R51) will mess this up anyway, because it will always pass when mode is ldaps, because the `@encryption` variable will never look like the string.

Reply:
@bdunne Your test wouldn't pass either because `@encyrption` is always at least an empty hash...I think better would be something like the following if we don't expect the key to be present at all.

```ruby
expect(ldap.ldap.instance_variable_get(:@encryption).try(:has_key_path?, :tls_options, :verify_mode)).to be_falsey
```

Reply:
Additionally since we own the MiqLdap library. can we just `attr_reader :encryption` for the sake of not having to do that instance_variable_get?

Reply:
Thank you @bdunne and @Fryguy 
I had gotten bogged testing this because, as @Fryguy points out, encryption variable is buried in the ldap object we don't own.

I'll update the spec to use @Fryguy suggestion